netmaster/objApi/apiController.go

```diff
@@ -1571,6 +1571,55 @@ func (ac *APIController) RuleCreate(rule *contivModel.Rule) error {
 		return core.Errorf("Policy not found")
 	}
 
+	if rule.Direction == "in" && rule.ToIpAddress != "" {
+		// rules from k8s network policy
+		// verify 'toIpAddress' is part of epg and subnet
+		if len(policy.LinkSets.EndpointGroups) != 1 {
+			errMsg := fmt.Errorf("failed to configure %s, %d endpoint groups linked to policy %s ",
+				rule.ToIpAddress,
+				len(policy.LinkSets.EndpointGroups),
+				rule.PolicyName)
+			log.Error(errMsg)
+			return errMsg
+		}
+		epgKey := ""
+		for key := range policy.LinkSets.EndpointGroups {
+			epgKey = strings.Replace(key, rule.TenantName+":", "", 1) + ":" + rule.TenantName
+		}
+
+		stateDriver, err := utils.GetStateDriver()
+		if err != nil {
+			log.Errorf("failed to configure %s, %s", rule.ToIpAddress, err)
+			return fmt.Errorf("failed to configure %s, unable to connect to key-value store",
```

Comment:
Curious why `key-value store` here? Is `stateDriver` stored as a key-value pair in etcd?